### ISSUE TITLE
[Blazor] The title should be something like Security considerations for Blazor Hybrid applications

### ISSUE THREAD

**Issue by javiercn** [dotnet/AspNetCore.Docs#25854]
The title should be something like `Security considerations for Blazor Hybrid applications` not `Authentication and Authorization` since the content doesn't have anything to do with that.

In addition to that, there is a bunch of content that we wanted to add here. Feel free to adapt it/compose it as you see fit with the existing content.

Blazor hybrid applications run .NET code in the application that renders web content inside a platform webview. The .NET code interacts with the web content via an interop channel provided between the two.
```mermaid
flowchart
  subgraph DotNet ["Application"]
    A[".NET Code"] <-- Interop -->
    B["WebView"]
  end
```

The web content rendered into the webview can come from assets provided by the app inside the wwwroot folder, or as it is the case in the web, it can come from an external source.

This creates a trust boundary between the .NET Code and the code that runs inside the webview. .NET Code comes from your application and the third-party packages you decide to trust and install, but once the app has been built, that can't change.

In contrast, content from the webview can come not only from your app, but as we said, from an external source. This can be the case for example if you are using a script from a CDN.

For this reason it is recommended to consider the code inside the webview as untrusted in a similar way you would do for code running inside the browser on a web application.

The same threats and recommendations apply on hybrid applications. If possible, avoid loading content from a third-party origin. To this effect you might be able to serve the content directly from within your app by downloading the assets to your wwwroot folder and packaging them with the rest of the app.

If your app has the requirement to reference content from an external origin it is recommended that you use common web security measures like HTTPS, a CSP policy, subresource integrity, etc. to make sure that if any of those resources gets compromised your app has a chance to block the resource from loading.

Even if you pack all the resources as part of the application and don't load any resource from an external origin, you still need to worry about problems in the code from those resources running inside the document as it might have vulnerabilities that might allow XSS attacks.

In general, Blazor protects against XSS by dealing with HTML in safe ways. However, there are patterns that allow components to inject raw HTML within Blazor itself and any JavaScript library that the application might be using using might also manipulate HTML in an unsafe way.

If your app manipulates HTML in an unsafe way either through Blazor or a third-party JavaScript library, your app might be vulnerable to XSS if for example it renders content coming from an untrusted source, like data coming from a database, etc.

For this reason, it is best to apply the same protections against XSS that you would do in a web application. Prevent loading scripts coming from unknown sources as well as potentially unsafe JavaScript features (like eval) via a content security policy.

If the code inside the webview is compromised, it will have access to all the content inside the document and might interact with the host via the interop channel. For that reason, any content coming from the document (events, JS interop) needs to be treated as untrusted and validated in the same way it is done in other sensitive contexts like Blazor Server.

It is also for this reason that it is considered bad practice to store sensitive information like credentials in the context of the webview, as it makes those available to an attacker if the webview is compromised when there are safer alternatives like handling them directly on the native portion of the application.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cb3ca917-28cc-126f-a7ff-3ff00ad30677
* Version Independent ID: cb3ca917-28cc-126f-a7ff-3ff00ad30677
* Content: [ASP.NET Core Blazor Hybrid authentication and authorization](https://docs.microsoft.com/en-us/aspnet/core/blazor/hybrid/security/?view=aspnetcore-6.0)
* Content Source: [aspnetcore/blazor/hybrid/security/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/hybrid/security/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by danroth27** [comment]
> The title should be something like Security considerations for Blazor Hybrid applications not Authentication and Authorization since the content doesn't have anything to do with that.

@javiercn The terms "authentication" and "authorization" appear repeatedly in this doc. If you're just looking for a place to put general security guidance for Blazor Hybrid we can create multiple docs under the Security node. Would that work?

**Comment by guardrex** [comment]
It seems like the content that you just supplied, @javiercn, is more general than the specifics on auth identity provider implementation. Perhaps, this new content will be the bulk of the introductory remarks for the overview. The detailed implementation guidance can be moved to a separate topic under the overview. For now, I'll place the new content at the top (I'm working on it right now, Sat 9:30am CST). You guys can let me know Sunday/Monday where the splits make sense and what goes where.

**Comment by javiercn** [comment]
@guardrex I think there might have been a mis understanding here?

This is the content I'm referring to

![image](https://user-images.githubusercontent.com/6995051/168465348-9dafd97f-0b4d-43e1-a0e6-00453f44fea1.png)
![image](https://user-images.githubusercontent.com/6995051/168465377-248162d7-db38-4e3c-b2e4-a8dbad89f123.png)

To be very clear. I would expect there to be two separate pages/articles:
* Authentication and authorization in Blazor Hybrid applications, including the content that I provided last week and that covers how to create an authentication state provider, etc.
* Security considerations for Blazor Hybrid applications. Which has the content referenced above and expands it with the content I provided.

Does that make sense?
